packages/vite/src/node/config.ts

```diff
@@ -757,22 +915,22 @@ function resolveEnvironmentResolveOptions(
   let conditions = resolve?.conditions
   conditions ??=
     consumer === 'client' || isSsrTargetWebworkerEnvironment
-      ? DEFAULT_CONDITIONS.filter((c) => c !== 'node')
-      : DEFAULT_CONDITIONS.filter((c) => c !== 'browser')
+      ? configDefaults.resolve.conditions.filter((c) => c !== 'node')
+      : configDefaults.resolve.conditions.filter((c) => c !== 'browser')
```

Comment:
It might be confusing that `configDefaults.resolve.conditions` contains both `node` and `browser`.

Reply:
Maybe we need `serverConfigDefaults.resolve.conditions`? This is a hard one. I think as you say having both values would be confusing.

Or maybe we should omit this one. And treat it as inferred. We could expose some defaults in a special way?
`clientDefaults.resolve.conditions`, `serverDefaults.resolve.conditions`?

Reply:
I removed `mainFields` and `conditions` from `configDefaults` and exposed the values separately. I think `clientDefaults` / `serverDefaults` might be confusing as it is also used for `ssr.target === 'webworker'`. If we remove all `isSsrTargetWebworkerEnvironment` in future, I think we can have `clientDefaults` / `serverDefaults` that contains all values that is applied conditionally by `consumer`.